Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,151 @@
+the hardware. A key requirement for such a split is that there is no dependency
+on a physical bus, device, register accesses or regmap support. These
+individual devices split from the core cannot live on the platform bus as they
+are not physical devices that are controlled by DT/ACPI. The same argument
+applies for not using MFD in this scenario as it relies on individual function
+devices being physical devices that are DT enumerated.
+
+An example for this kind of requirement is the audio subsystem where a single
+IP handles multiple entities such as HDMI, Soundwire, local devices such as
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The virtual bus is intended to be minimal, generic and avoid domain-specific
+assumptions. Each virtual bus device represents a part of its parent
+functionality. The generic behavior can be extended and specialized as needed
+by encapsulating a virtual bus device within other domain-specific structures
+and the use of .ops callbacks. Devices on the same virtual bus do not share any
+structures and the use of a communication channel with the parent is
+domain-specific.
+
+Virtbus Devices
+~~~~~~~~~~~~~~~
+
+A virtbus_device is created and registered to represent a part of its parent
+device's functionality. It is given a match_name that is used for driver
+binding and a release callback that is invoked when the device is unregistered.
+
+.. code-block:: c
+
+	struct virtbus_device {
+		struct device dev;
+		const char *match_name;
+		void (*release)(struct virtbus_device *);
+		u32 id;
+	};
+
+The virtbus device is enumerated when it is attached to the bus. The device
+is assigned a unique ID automatically that will be appended to its name. If
+two virtbus_devices both named "foo" are registered onto the bus, they will
+have the device names, "foo.x" and "foo.y", where x and y are unique integers.
+
+Virtbus Drivers
+~~~~~~~~~~~~~~~
+
+Virtbus drivers follow the standard driver model convention, where
+discovery/enumeration is handled by the core, and drivers
+provide probe() and remove() methods. They support power management
+and shutdown notifications using the standard conventions.
+
+.. code-block:: c
+
+	struct virtbus_driver {
+		int (*probe)(struct virtbus_device *);
+		int (*remove)(struct virtbus_device *);
+		void (*shutdown)(struct virtbus_device *);
+		int (*suspend)(struct virtbus_device *, pm_message_t);
+		int (*resume)(struct virtbus_device *);
+		struct device_driver driver;
+		const struct virtbus_device_id *id_table;
+	};
+
+Virtbus drivers register themselves with the bus by calling
+virtbus_register_driver(). The id_table contains the names of virtbus devices
+that a driver can bind with?
```

Comment:
why a question mark? A typo?

Reply:
I think it was my suggestion with a question mark that was copied verbatim...

Reply:
yes, sorry about that. The documentation is so verbose now that I missed it :(